pkg/bff/auth/auth.go

```diff
@@ -0,0 +1,207 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware/v2"
+	"github.com/auth0/go-jwt-middleware/v2/jwks"
+	"github.com/auth0/go-jwt-middleware/v2/validator"
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+	"github.com/trisacrypto/directory/pkg/bff/api/v1"
+	"github.com/trisacrypto/directory/pkg/bff/config"
+)
+
+const (
+	ScopeAnonymous          = "anonymous"
+	ContextUserInfo         = "auth0_user_info"
+	ContextBFFClaims        = "auth0_bff_claims"
+	ContextRegisteredClaims = "auth0_registered_claims"
+)
+
+// AnonymousClaims are used to identify unauthenticated requests that have no permissions.
+var AnonymousClaims = Claims{Scope: ScopeAnonymous, Permissions: nil}
+
+// Claims extracts custom data from the JWT token provided by Auth0
+type Claims struct {
+	Scope       string   `json:"scope"`
+	Permissions []string `json:"permissions"`
+	VASP        string   `json:"https://vaspdirectory.net/vasp"`
+	Email       string   `json:"https://vaspdirectory.net/email"`
+}
```

Comment:
Scope comes by default on the access token from Auth0. The Permissions will come if you tick a check mark in the settings of the API. The `VASP` and `Email` claims required me to create Auth0 actions to add them to the access token.